apps/client/src/pages/home.tsx

```diff
@@ -72,31 +69,61 @@ export default function HomePage(): JSX.Element {
                 const identity = new Identity(await signer.signMessage(message))
                 const identityCommitment = identity.getCommitment().toString()
 
-                const hasJoined = await isGroupMember(
-                    invite.group.id,
-                    identityCommitment
-                )
+                if (invite.group.type === "off-chain") {
+                    const hasJoined = await bandadaAPI.isGroupMember(
+                        invite.group.id,
+                        identityCommitment
+                    )
 
-                if (hasJoined === null) {
-                    setLoading(false)
-                    return
-                }
+                    if (hasJoined === null) {
+                        setLoading(false)
+                        return
+                    }
 
-                if (hasJoined) {
-                    setLoading(false)
-                    alert("You have already joined this group")
-                    return
-                }
+                    if (hasJoined) {
+                        setLoading(false)
+                        alert("You have already joined this group")
+                        return
+                    }
 
-                const response = await addMemberByInviteCode(
-                    invite.group.id,
-                    identityCommitment,
-                    inviteCode
-                )
+                    const response = await bandadaAPI.addMemberByInviteCode(
+                        invite.group.id,
+                        identityCommitment,
+                        inviteCode
+                    )
 
-                if (response === null) {
-                    setLoading(false)
-                    return
+                    if (response === null) {
+                        setLoading(false)
+                        return
+                    }
+                } else {
+                    const group = await semaphoreAPI.getGroup(invite.group.name)
```

Comment:
After the invite code is used, users can use it again because it was not marked as used. Something we can do is to create an API SDK function for this: https://github.com/bandada-infra/bandada/blob/main/apps/api/src/app/invites/invites.service.ts#L126 

(for that we need to add that function to the controller to create a function in the API SDK)

And call that new api sdk function from the client app.

The idea is to have a similar devex as invite code for off-chain groups when using this new invite code for on-chain groups.

More context about the `redeemInvite` function: https://github.com/bandada-infra/bandada/blob/main/apps/api/src/app/groups/groups.service.ts#L425

Does it make sense? 

As a note to keep in mind for developers when using this functionality is that these actions with invite code for on-chain groups should be done in backend in a real world app because of security issues since users will be able to join a group even if the invite is not correct. But I think it's fine to keep it here in the client app browser for now as a way to showcase the functionality. It should be moved to a serverless function later in the future.

Reply:
Correct me if I get it wrong, the new api-sdk is to bundle `redeemInvite` together with add member to group. 

Depends on whether the invite is for on-chain or off-chain, if off-chain, the api-sdk will add the member into the off-chain group; if on-chain, the api-sdk will run Semaphore `addMember`.

Did I miss anything?

Reply:
I just merged this PR because this could be part of a new PR.

The idea is to add just one function to the API SDK. The function can be called `redeemInvite` and would go here: https://github.com/bandada-infra/bandada/blob/main/libs/api-sdk/src/invites.ts

Then we would call this function in the client app before adding the member with Semaphore. 

The other part related to improving devex when using invites for on-chain groups may be part of a different PR.

Reply:
> In cases of users failed to join the group upon clicking the join group button, should we invalidate the invite code or leave it until the invite has been redeemed successfully?

I think wait until the transaction is complete.

Reply:
Oh so the new API SDK is just to `redeemInvite`? So if developers did not use the `redeemInvite` API SDK, they could reuse the invite code?

Reply:
> Oh so the new API SDK is just to redeemInvite?

Yes, correct.

> So if developers did not use the redeemInvite API SDK, they could reuse the invite code?

Yes, if `redeemInvite` (in the Bandada backend) is never called, the invite is still valid. That `redeemInvite` function is called in the Bandada infrastructure (backend) when a user joins a group using an invite code, this way the invite code can't be used more than once.

Reply:
Got it, thanks for the clarifications! Will work on it!

Reply:
You're welcome. Thank you very much 🙏